src/examples/workflo/macros/examples/screen_app.cljs

```diff
@@ -0,0 +1,268 @@
+                       :onChange #(.update this :user/email %)})))))
+
+(defview UserView
+  [db [id] user [name email]]
+  (commands [goto])
+  (render
+   (dom/div nil
+     (dom/p nil (str name " <" email ">"))
+     (dom/p nil
+       (dom/button #js {:onClick #(goto {:screen 'UserScreen
+                                         :params {:user-id id}})}
+         "View")
+       (dom/button #js {:onClick #(goto {:screen 'UserSettingsScreen
+                                         :params {:user-id id}})}
+         "Settings")))))
+
+(defview UserSettingsTitleView
+  [db [id] user [name]]
+  (render
+   (dom/h2 nil (str "Settings for " id ": " name))))
+
+(defview UserSettingsTitle
+  [({user UserSettingsTitleView} {db/id ?user-id})]
+  (render
+   (user-settings-title-view user)))
+
+(defview UserSettings
+  [({user UserSettingsView} {db/id ?user-id})]
+  (key (:db/id (:user props)))
+  (render
+   (user-settings-view user)))
+
+(defview UserTitleView
+  [db [id] user [name]]
+  (render
+   (dom/h2 nil (str "User " id ": " name))))
+
+(defview UserTitle
+  [({user [user [name]]} {db/id ?user-id})]
+  (render
+   (user-title-view user)))
+
+(defview UserProfile
+  [({user UserView} {db/id ?user-id})]
+  (render
+   (user-view user)))
+
+(defview UserListTitle
+  (render
+   (dom/h2 nil "User List")))
+
+(defview UserList
+  [{users UserView}]
+  (commands [add-user])
+  (render
+   (dom/div nil
+     (for [u users]
+       (user-view u))
+     (dom/p nil
+       (dom/button #js {:onClick #(add-user {:db/id (inc (count users))
+                                             :user/name ""
+                                             :user/email ""})}
+         "Add user")))))
+
+;;;; Screens
+
+(defscreen UserSettingsScreen
+ (url "users/:user-id/settings")
+ (navigation
+  (title "User Settings"))
+ (layout
+  {:title {:view UserSettingsTitle :factory user-settings-title}
+   :content {:view UserSettings :factory user-settings}}))
+
+(defscreen UserScreen
+  (url "users/:user-id")
+  (navigation
+   (title "User"))
+  (layout
+   {:title {:view UserTitle :factory user-title}
+    :content {:view UserProfile :factory user-profile}}))
+
+(defscreen UserListScreen
+  (url "users")
+  (navigation
+    (title "Users"))
+  (layout
+   {:title {:view UserListTitle :factory user-list-title}
+    :content {:view UserList :factory user-list}}))
+
+;;;; Commands
+
+(s/def :db/id (s/and int? pos?))
+(s/def :user/name (s/or :nil nil? :string string?))
+(s/def :user/email (s/or :nil nil? :string string?))
+(s/def ::user
+  (s/keys :req [:db/id]
+          :opt [:user/name
+                :user/email]))
+
+(defcommand add-user
+  [::user]
+  (do
+    (println "add-user" data)
+    {:state {(:db/id data) data}
+     :location {:screen 'UserSettingsScreen
+                :params {:user-id (:db/id data)}}}))
+
+(defcommand update-user
+  [::user]
+  (do
+    (println "update-user" data)
+    {:state {(:db/id data) data}}))
```

Comment:
We don't have that service yet and in the example app we'll perhaps call it something else (`db` or `datascript`).